controllers/users.controller.js

```diff
@@ -90,20 +85,129 @@ const updateUserDetails = async (req, res) => {
 				lastname: user.lastname,
 				userId: user._id,
 			},
-			success: true,
 		});
 	} catch (error) {
 		res.status(500).json({
-			success: false,
 			message: 'Request failed please check errorMessage key for more details',
 			errorMessage: error.message,
 		});
 	}
 };
 
+const getUserByIdFromDb = async (req, res) => {
+	try {
+		const { userId } = req.params;
+		console.log(userId);
+		const user = await User.findById({ _id: userId });
+
+		if (!user) {
+			res.status(404).json({ message: 'email does not exist!' });
+			return;
+		}
+
+		res.status(200).json({
+			response: {
+				email: user.email,
+				firstname: user.firstname,
+				lastname: user.lastname,
+				userId: user._id,
+			},
+		});
+	} catch (error) {
+		res.status(500).json({
+			message: 'Request failed please check errorMessage key for more details',
+			errorMessage: error.message,
+		});
+	}
+};
+
+const getNotesOfVideoOfUserFromDb = async (req, res) => {
+	try {
+		const { userId, videoId } = req.params;
+		const notes = await Note.find({ userId: userId, videoId: videoId });
+		res.status(200).json({ response: notes });
+	} catch (error) {
+		res.status(500).json({
+			message: 'Something went wrong',
+			errorMessage: error.message,
+		});
+	}
+};
+
+const getOrCreatePlaylistsOfUser = async (req, res) => {
+	try {
+		const { userId } = req.params;
+
+		let playlists = await Playlist.find({ userId }).populate({
+			path: 'videoList.videoId',
+			populate: { path: 'tutorId' },
+		});
+
+		if (playlists.length === 0) {
+			let watchlaterPlaylist = new Playlist({
+				userId,
+				type: 'watchlater',
+				isDefault: true,
+				videoList: [],
+			});
+			let historyPlaylist = new Playlist({
+				userId,
+				type: 'history',
+				isDefault: true,
+				videoList: [],
+			});
+			let likedPlaylist = new Playlist({
+				userId,
+				type: 'liked',
+				isDefault: true,
+				videoList: [],
+			});
+			await Playlist.insertMany([
+				watchlaterPlaylist,
+				historyPlaylist,
+				likedPlaylist,
+			]);
```

Comment:
I think the default playlists should be created as soon as a new user is created

Reply:
Yes, that also works. But actually I am using the same user collection for all the apps, so I want to create playlists only if that user visits video library app. Hence, not creating it on sign up.